### PR TITLE
Fixed issue where `hsmPlatform` was not being set in `KeyProperties`.

### PR DESCRIPTION
**Pull request by vcolin7**

### CODE REVIEW

**Review by mccoyp — APPROVED** [review]
Looks good to me based on the implementation of other properties that were added in later API versions, like `exportable` -- though I can't comment on Java implementation specifics 🙂

**Review by alzimmermsft — APPROVED** [review]
Was this something missing in the Swagger definition leading to it not being generated into the implementation `KeyAttributes`?

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java [hunk 1]
```diff
@@ -69,7 +69,10 @@ public void createKey(HttpClient httpClient, KeyServiceVersion serviceVersion) {
 
         createKeyRunner((keyToCreate) ->
             StepVerifier.create(keyAsyncClient.createKey(keyToCreate))
-                .assertNext(response -> assertKeyEquals(keyToCreate, response))
+                .assertNext(createdKey -> {
+                    assertKeyEquals(keyToCreate, createdKey);
+                    assertNotNull(createdKey.getProperties().getHsmPlatform());
```
Comment (gearama) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java:74]: should we have a check for the actual value also ?

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java [hunk 2]
```diff
@@ -205,11 +208,17 @@ public void getKey(HttpClient httpClient, KeyServiceVersion serviceVersion) {
 
         getKeyRunner((keyToSetAndGet) -> {
             StepVerifier.create(keyAsyncClient.createKey(keyToSetAndGet))
-                .assertNext(response -> assertKeyEquals(keyToSetAndGet, response))
+                .assertNext(createdKey -> {
+                    assertKeyEquals(keyToSetAndGet, createdKey);
+                    assertNotNull(createdKey.getProperties().getHsmPlatform());
```
Comment (gearama) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java:213]: should we have a check for the actual value also ?

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java [hunk 3]
```diff
@@ -205,11 +208,17 @@ public void getKey(HttpClient httpClient, KeyServiceVersion serviceVersion) {
...
+                })
                 .verifyComplete();
 
             StepVerifier.create(keyAsyncClient.getKey(keyToSetAndGet.getName()))
-                .assertNext(response -> assertKeyEquals(keyToSetAndGet, response))
+                .assertNext(retrievedKey -> {
+                    assertKeyEquals(keyToSetAndGet, retrievedKey);
+                    assertNotNull(retrievedKey.getProperties().getHsmPlatform());
```
Comment (gearama) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java:220]: should we have a check for the actual value also ?

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java [hunk 4]
```diff
@@ -66,7 +66,12 @@ protected void createKeyClient(HttpClient httpClient, KeyServiceVersion serviceV
     public void createKey(HttpClient httpClient, KeyServiceVersion serviceVersion) {
         createKeyClient(httpClient, serviceVersion);
 
-        createKeyRunner((keyToCreate) -> assertKeyEquals(keyToCreate, keyClient.createKey(keyToCreate)));
+        createKeyRunner((keyToCreate) -> {
+            KeyVaultKey createdKey = keyClient.createKey(keyToCreate);
+
+            assertKeyEquals(keyToCreate, createdKey);
+            assertNotNull(createdKey.getProperties().getHsmPlatform());
```
Comment (gearama) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java:73]: should we have a check for the actual value also ?

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java [hunk 5]
```diff
@@ -184,7 +189,10 @@ public void getKey(HttpClient httpClient, KeyServiceVersion serviceVersion) {
         getKeyRunner((keyToSetAndGet) -> {
             keyClient.createKey(keyToSetAndGet);
 
-            assertKeyEquals(keyToSetAndGet, keyClient.getKey(keyToSetAndGet.getName()));
+            KeyVaultKey retrievedKey = keyClient.getKey(keyToSetAndGet.getName());
+
+            assertKeyEquals(keyToSetAndGet, retrievedKey);
+            assertNotNull(retrievedKey.getProperties().getHsmPlatform());
```
Comment (gearama) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java:195]: should we have a check for the actual value also ?

**Review by gearama — COMMENTED** [review]
not the code owner , i did have some comments.

**Comment by vcolin7** [comment]
> Was this something missing in the Swagger definition leading to it not being generated into the implementation `KeyAttributes`?

No, when releasing the last stable version the library I did not re-generate it and just added fields to the custom models we expose to users.

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java (hunk 1 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java:74]: I just wanted to make sure we deserialize properly but we can check for the value. It should be `0` by default from what I've seen.

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java (hunk 2 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java:213]: See my other comment about this.

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java (hunk 3 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java:220]: See my other comment about this.

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java (hunk 4 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java:73]: See my other comment about this.

**Comment by alzimmermsft** [comment]
> > Was this something missing in the Swagger definition leading to it not being generated into the implementation `KeyAttributes`?
> 
> @alzimmermsft No, when releasing the last stable version the library I did not re-generate it and just added fields to the custom models we expose to users.

So, does that mean `hmsPlatform` is in the Swagger but was just missed last release? Or did that mean this is being added manually to a generated model?

If the former, good with me. If the latter, this is a spot that could be missed again and we should either get the value into the Swagger if everyone needs this or add a Swagger transform to add it into code generation.

**Comment by vcolin7** [comment]
> > > Was this something missing in the Swagger definition leading to it not being generated into the implementation `KeyAttributes`?
> > 
> > 
> > @alzimmermsft No, when releasing the last stable version the library I did not re-generate it and just added fields to the custom models we expose to users.
> 
> So, does that mean `hmsPlatform` is in the Swagger but was just missed last release? Or did that mean this is being added manually to a generated model?
> 
> If the former, good with me. If the latter, this is a spot that could be missed again and we should either get the value into the Swagger if everyone needs this or add a Swagger transform to add it into code generation.

The value is in the Swagger but I just manually updated the public APIs and did not remember to re-generate the impl classes.

**Pull request merged**